### DISCUSSION TITLE
Oprimize flow user experience.

### DISCUSSION THREAD

**Discussion by fanmingfei** [directus/directus#21737] [Feature Requests]
### Summary

Faced with complex flows, there's a need for simpler shortcuts to reduce operational costs.

#21723 

### Motivation

Faced with complex flows, there's a need for simpler shortcuts to reduce operational costs.

### Detailed Design

https://github.com/directus/directus/assets/4632277/b09c846f-79ea-4875-aa6d-349e96289514

https://github.com/directus/directus/assets/4632277/ff5c6cd9-3313-4f28-9385-bfa8131e73ee

### Requirements List

- Add a shortcut key for saving the operation
- You can directly double-click on the operation panel to edit
- Editing of the flow from the shortcut
- Refresh flow execution log when returning to the current tab page.

### Drawbacks

-

### Alternatives

-

### Adoption Strategy

-

**Comment by JayShoe** [comment]
Hello, 

Overall I agree with this ticket. Adding some quality of life things to the flows would be nice.  I have some things I'd like to see as well. Maybe if this request is voted we could consider adding some additional optimizations. 

### Add Links for Triggered From and Triggerd To Items
I created a separate issue for this, but I think it fits here. https://github.com/directus/directus/discussions/22092. The idea is to add links to any flow that may trigger the current flow, and also links to any flow that might be triggered from this flow. This allows easier multi-step flow management without having to go back to the flow list and select the flow you need to view. 

### Encase the Flow Payload with the Operation Name
**I consider this one HUGE.** When looking at the logs, it would be nice if the operation payload was encased in the operation name - as it would be in the actual results and won't leave me guessing (or referencing) what the operation name is. 

For example,
This 
`{
  "event": "customrecord_gfr_company.items.create",
  "payload": {
    "id": "250"
  },
  "key": "2175ce88-7994-4c92-9e26-ac4a0e481d54",
  "collection": "customrecord_company"
}`
Would now be this
`"$trigger": {
{
  "event": "customrecord_company.items.create",
  "payload": {
    "id": "250"
  },
  "key": "2175ce88-7994-4c92-9e26-ac4a0e481d54",
  "collection": "customrecord_gfr_company"
}
}`

This simple addition will make understanding flows MUCH easier for a newcomer. We will now understand visually what $trigger means, and for each and every operation, we will get a clear visual as to where the data we are seeking is located in the tree of JSON. 

### Show JSON in Code Block
By Placing the JSON in a code block we can more easily search it, and we can copy it. This should follow the interface for Code:JSON as in the current field interface functions. 

### Copy JSON LInks
I would like to simply copy the entire JSON block so I can view it or use it elsewhere easier. 

### (Alternate/Addition to Code Block) Collapse and Expand JSON Tree in Logs

Building off the last idea, I would love to see a tree view in the payloads of each flow operation's log. This will allow us to collapse the tree (items and sub-items), to visually inspect the JSON tree and understand it's structure more clearly. Unfortunately, this probably requires a library - which I'm sure is avoided. Here is an example of what I mean 
https://pgrabovets.github.io/json-view/ 
https://www.npmjs.com/package/vue-json-viewer
https://www.npmjs.com/package/vue3-json-viewer

### Tooltip over each variable

I would love to see a tooltip when hovering over a field name in the JSON tree that would show me the location of the variable. For example, when hovering over "id" in this example, you would view {{$trigger.payload.id}}. If you click on the item, it should copy that location to your clipboard. 

`"$trigger": {
{
  "event": "customrecord_company.items.create",
  "payload": {
    "id": "250"
  },
  "key": "2175ce88-7994-4c92-9e26-ac4a0e481d54",
  "collection": "customrecord_gfr_company"
}
}`

### Limit the count of logs returned
I've noticed when I've ran a view many times, the logs won't load very quickly. They seem to be loading hundreds of logs each time the Log button is pressed. Maybe limit the query to return only the first 10 and then call a "more" to view more.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
